DevelopedCalculator.js

```diff
@@ -2,25 +2,30 @@
 class DevelopedCalculator extends Calculator {
     // defining history as an array
     history = [this.currentResult];
-    // saving historyDiv as a property on the class
-    historyDiv = null;
+    // saving historyElement as a property on the class
+    historyElement = null;
     render() {
             //calling the render method of the parent class
             super.render()
                 // first create a div 
-            this.historyDiv = document.createElement("div");
+            this.historyElement = document.createElement("div");
             // add the text to the div
-            this.historyDiv.appendChild(document.createTextNode("History:" + this.currentResult));
-            // prepend the historyDiv in the host element 
-            this.hostElem.prepend(this.historyDiv);
+            this.historyElement.appendChild(document.createTextNode("History:" + this.currentResult));
+            // prepend the historyElement in the host element 
+            this.hostElem.prepend(this.historyElement);
+            // this.gettingHistoryArray();
         }
         // new method has a parameter
     onButtonNumberClick(event) {
         // calling parent class's onButtonNumberClick method
         super.onButtonNumberClick(event);
         // pushing the history array with current result
         this.history.push(this.currentResult);
+        this.gettingHistoryArray();
     }
+    gettingHistoryArray() {
+        this.history.join(",");
```

Comment:
`this.history.join(",");` returns the history items separated by ","
I can't see your method `gettingHistoryArray` doing anything useful.
despite this part of the code ``this.history.join(",");` is correct and needed, but you're not using that in anything 

you're not updating the historyDiv which you renamed to historyElement with returned value of `this.history.join(",");

can you read the requirements again pls?